File: 856-score-of-parentheses/856-score-of-parentheses.java
class Solution {
    public int scoreOfParentheses(String s) {
        Stack<Integer> st = new Stack<>();
        int cur = 0;
        for(char c: s.toCharArray()) {
            if(c=='(') {
                st.push(cur);
                cur = 0;
            }
            else {
                cur = st.pop() + Math.max(2*cur, 1);
            }
        }
        return cur;
    }
}

File: 856-score-of-parentheses/856-score-of-parentheses.cpp
class Solution {
public:
    int scoreOfParentheses(string s) {
        stack<string> st;
        for(int i=0;i<s.length();i++) {
            char c = s[i];
            if(st.empty()) {
                string t;
                t+=c;
                st.push(t);
            }
            else {
                if(c==')') {
                    int curScore = 0;
                    while(!st.empty() && st.top()!="(") {
                        curScore += stoi(st.top());
                        st.pop();
                    }
                    st.pop();
                    if(curScore==0)
                        st.push("1");
                    else 
                        st.push(to_string(2*curScore));
                }
                else {
                    string t;
                    t+=c;
                    st.push(t);
                } 
            }
        }
        int score = 0;
        while(!st.empty()) {
            score += stoi(st.top());
            st.pop();
        }
        return score;
    } 
};